### PR TITLE
[batch] Add requester pays option

### PR DESCRIPTION
**Pull request by jigold**
I debated whether to have this option on a batch which would require a database migration to add a new field to the batches table or just have it in the job spec for all jobs (same for all jobs). What I did is the easiest thing. It assumes all requester pay accesses are billed to the same project.

### CODE REVIEW

**Comment by jigold** [comment]
I also made a new test bucket specifically for the tests -- gs://hail-services-requester-pays/

**Comment by jigold** [comment]
@danking I'm getting this error. Do you see any problem with granting that capability to the test service account?

```
+ retry gcloud -q auth activate-service-account '--key-file=/gsa-key/key.json'
+ gcloud -q auth activate-service-account '--key-file=/gsa-key/key.json'
Activated service account credentials for: [test-665@hail-vdc.iam.gserviceaccount.com]
+ mkdir -p /io/batch/27b395/inputs/wjDTI
+ retry gsutil -u hail-vdc -m cp -R gs://hail-services-requester-pays/hello /io/batch/27b395/inputs/wjDTI/hello
+ gsutil -u hail-vdc -m cp -R gs://hail-services-requester-pays/hello /io/batch/27b395/inputs/wjDTI/hello
AccessDeniedException: 403 test-665@hail-vdc.iam.gserviceaccount.com does not have serviceusage.services.use access to the Google Cloud project.
CommandException: 1 file/object could not be transferred.
+ sleep 2
+ gsutil -u hail-vdc -m cp -R gs://hail-services-requester-pays/hello /io/batch/27b395/inputs/wjDTI/hello
AccessDeniedException: 403 test-665@hail-vdc.iam.gserviceaccount.com does not have serviceusage.services.use access to the Google Cloud project.
CommandException: 1 file/object could not be transferred.
+ sleep 5
+ gsutil -u hail-vdc -m cp -R gs://hail-services-requester-pays/hello /io/batch/27b395/inputs/wjDTI/hello
AccessDeniedException: 403 test-665@hail-vdc.iam.gserviceaccount.com does not have serviceusage.services.use access to the Google Cloud project.
CommandException: 1 file/object could not be transferred.
```

**Comment by danking** [comment]
Seems like the correct privilege to get the test account so that it can use requestor pays.

File: hail/python/hailtop/batch/batch.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ class Batch:
     attributes: :obj:`dict` of :obj:`str` to :obj:`str`, optional
         Key-value pairs of additional attributes. 'name' is not a valid keyword.
         Use the name argument instead.
+    requester_pays: :obj:`str`, optional
```
Comment (danking) [inline hail/python/hailtop/batch/batch.py:53]: Can we pervasively use a more descriptive name? Maybe `requester_pays_project`?

**Pull request merged**